app/services/queue_listener.rb

```diff
@@ -40,13 +41,20 @@ def perform(sqs_message, params)
       when UPDATE_MEMBER
         update_member(params)
 
+      when SUBSCRIPTION_CANCELLATION
+        cancel_subscription(params[:params])
+
       else
         raise ArgumentError, "Unsupported message type: #{params[:type]}"
     end
   end
 
   private
 
+  def cancel_subscription(params)
+    client.cancel_subscription(params)
```

Comment:
I'm working on something related to this so I can add the check for errors now (I need this to work for sure for the reactivation emails, so checking for a success response was on the todo list).